File: app/src/main/java/com/lucasurbas/counter/app/di/vm/ViewModelModule.java
package com.lucasurbas.counter.app.di.vm;

import android.arch.lifecycle.ViewModelProvider;

import dagger.Binds;
import dagger.Module;

@Module
public abstract class ViewModelModule {

    @Binds
    abstract ViewModelProvider.Factory provideViewModelFactory(ViewModelFactory viewModelFactory);
}

File: app/src/main/java/com/lucasurbas/counter/app/di/vm/ViewModelModule.kt
package com.lucasurbas.counter.app.di.vm

import android.arch.lifecycle.ViewModelProvider

import dagger.Binds
import dagger.Module

@Module
abstract class ViewModelModule {

    @Binds
    internal abstract fun provideViewModelFactory(viewModelFactory: ViewModelFactory): ViewModelProvider.Factory
}
